src/data/list/erase_dup.lean

```diff
@@ -5,35 +5,37 @@ Authors: Mario Carneiro
 -/
 import data.list.nodup
 
-universes u
-variables {α : Type u}
+/-!
+# Erasure of duplicates in a list
 
-namespace list
-
-/- erase duplicates function -/
+This file proves basic results about `list.erase_dup` (definition in `data.list.defs`).
+`erase_dup l` returns `l` without its duplicates. It keeps the earliest (that is, rightmost)
+occurrence of each.
+-/
 
-variable [decidable_eq α]
+namespace list
+variables {α : Type*} [decidable_eq α]
```

Comment:
There is a slight advantage to using `Type u` over `Type*`. IIRC `Type u` says that the universe is a free variable, while `Type*` leaves it a metavariable, that gets promoted to a free variable at the end of elaboration. The distinction (probably) does not matter at all here, I'm just a bit wary of introducing regressions in a lintfix PR.
```suggestion
universes u

namespace list

variables {α : Type u} [decidable_eq α]
```